public/javascripts/main.js

```diff
@@ -64,9 +125,20 @@ const ArticlesList = React.createClass({
 });
 
 const ArticlesBox = React.createClass({
+
+    resetPage() {
+        this.props.page = 0;
+    },
+    componentDidMount() {
+        window.addEventListener('scroll', this.handleScroll);
```

Comment:
I like the idea of the infinite scroll but the `scroll` event is called way too often to be a proper solution. This is the evolution of the pagination variables back-end side:
![pagination](https://cloud.githubusercontent.com/assets/3872125/26776526/635a2ece-49d1-11e7-9eca-a78a1ac943c3.gif)